### PR TITLE
Adding PermissionsBoundary property for State Machine resource

### PR DESCRIPTION
**Pull request by vaib-amz**
*Issue #, if available:*

*Description of changes:*

* Adding `PermissionsBoundary` property for State Machine resource similar to that for [SAM Function](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html#sam-function-permissionsboundary)
* `PermissionsBoundary` property only works if the role is generated by SAM (using the `Policies` property)
* `PermissionsBoundary` property also applies to the roles created for the event sources.

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [x] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by c2tarun — APPROVED** [review]
LGTM!

File: samtranslator/model/stepfunctions/events.py [hunk 1]
```diff
@@ -42,8 +42,8 @@ def _generate_logical_id(self, prefix, suffix, resource_type):
             logical_id = generator.gen()
         return logical_id
 
-    def _construct_role(self, resource, prefix=None, suffix=""):
-        """Constructs the IAM Role resource allowing the event service to invoke
+    def _construct_role(self, resource, permissions_boundary=None, prefix=None, suffix=""):
+        """Constructs the IAM Role resource allowing the event service to invoke 
         the StartExecution API of the state machine resource it is associated with.
 
         :param model.stepfunctions.StepFunctionsStateMachine resource: The state machine resource associated with the event
```
Comment (hawflau) [inline samtranslator/model/stepfunctions/events.py:49]: Can you update the docstring for the new param as well?

File: samtranslator/model/stepfunctions/generators.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ def __init__(
         logging,
         name,
         policies,
+        permissions_boundary,
```
Comment (hawflau) [inline samtranslator/model/stepfunctions/generators.py:40]: Same here. Can you update the docstring for the new param as well?

**Comment by wong-a** [comment]
I'm taking over this work for @vaib-amz. @hawflau @c2tarun I addressed your comments on this PR, which just cherry-picks the change and adds the docstrings. 

https://github.com/aws/serverless-application-model/pull/1772

**Review by c2tarun — APPROVED** [review]

**Comment by hawflau** [comment]
Closing as #1772 is merged

**Pull request closed without merging**